blocks/embed/embed.js

```diff
@@ -0,0 +1,222 @@
+        box-shadow: none; display: block;" loading="lazy">
+      </iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedVimeo = (url) => {
+    const linkArr = url.href.split('/');
+    const video = linkArr ? linkArr[3] : linkArr;
+
+    const embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+        <iframe src=\"https:\/\/player.vimeo.com\/video\/${video}?app_id=122963\" 
+        style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" 
+        frameborder=\"0\" allow=\"autoplay; fullscreen; picture-in-picture\" allowfullscreen  
+        title="Content from Vimeo" loading="lazy"><\/iframe>
+    </div>`;
+    return embedHTML;
+}
+  
+const embedSpark = (url) => {
+    let embedURL = url;
+    if (!url.pathname.endsWith('/embed.html') && !url.pathname.endsWith('/embed')) {
+      embedURL = new URL(`${url.href}${url.pathname.endsWith('/') ? '' : '/'}embed.html`);
+    }
+  
+    return getDefaultEmbed(embedURL);
+}
+  
+const embedTwitter = (url) => {
+    const embedHTML = `<blockquote class="twitter-tweet"><a href="${url}"></a></blockquote>`;
+    loadScript('https://platform.twitter.com/widgets.js');
+    return embedHTML;
+}
+
+const embedTiktok = (url) => {
+    let resultHtml = document.createElement('div');
+    resultHtml.setAttribute('id', 'tiktok');
+    fetch(`https://www.tiktok.com/oembed?url=${url}`)
+    .then(response => response.json())
+    .then(data => {
+        loadScript('https://www.tiktok.com/embed.js');
+        const $tiktok = document.getElementById('tiktok')
+        $tiktok.outerHTML = data.html;
+    })
+    .catch((error) => {
+        console.error('Error:', error);
+    });
+
+    return resultHtml.outerHTML;
+}
+
+const embedSlidShare = (url) => {
+    let resultHtml = document.createElement('div');
+    resultHtml.setAttribute('id', 'slideShare');
+    fetch(url)
+    .then(data => {
+        data.text().then( data => {
+            const $el = document.createElement('div');
+            $el.innerHTML = data;
+            const embedUrl = $el.querySelector('.slideshow-info meta[itemprop="embedURL"]').content;
+            if(embedUrl) {
+                const $slideShare = document.getElementById('slideShare')
+                $slideShare.outerHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+                <iframe src="${embedUrl}" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" allowfullscreen loading="lazy"> </iframe>
+                </div>`;
+            }
+        });
+    })
+    .catch((error) => {
+        console.error('Error:', error);
+    });
+    
+    return resultHtml.outerHTML;
+}
+  
+const EMBEDS_CONFIG = {
```

Comment:
this data structure seems a bit redundant. i think in an ideal world the structure would probably look closer to:

```
{
  vimeo: {
    hostnames: [ 'www.vimeo.com', 'vimeo.com', 'player.vimeo.com'],
    embed: embedVimeo,
  }
}
```

Reply:
i definitely like your simplification a lot, but i think there are still a couple of special cases like links to `youtu.be` which as used commonly which might fall through the cracks here or need some special treatment (in js).

so i think there are two paths from here:
(1) either the data structure accommodates for all the hostnames as the one above and we will probably not need any special casing in JS or
(2) you could simplify the current data structure to:
```
{
  'youtube': embedYoutube,
  'vimeo': embedVimeo,
 }

```